templates/frontend/pages/announcements.tpl

```diff
@@ -0,0 +1,27 @@
+{**
+ * templates/frontend/pages/announcements.tpl
+ *
+ * Copyright (c) 2013-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * About the Journal Subscriptions.
+ *
+ *}
+
+{extends file="layouts/backend.tpl"}
+{block name="page"}
+    <pkp-header></pkp-header>
+
+    <h1 class="app__pageHeading">
+        {translate key=$pageTitle}
+    </h1>
+
+    <ul>
+        {foreach from=$announcements item=announcement}
+            <li style="margin: 8px 0;">
+                {$announcement->getId()}  / {$announcement->getLocalizedData('title')} / {$announcement->getDatePosted()} / {$announcement->getLocalizedData('keyword')} / <pkp-button element="a" href="http://localhost:8000/index.php/first-journal/exercise/announcements/{$announcement->getId()|strip_unsafe_html}" is-link>View</pkp-button>
```

Comment:
Be careful with escaping; the announcement title is expected to be plain text, so it should be passed through the `|escape` Smarty modifier to make sure that we aren't vulnerable to XSS leaks, and that `<` / `>` / `&` characters in announcement titles are appropriately presented. Generally it's a good idea to `|escape` just about everything; there are only a few cases where it's not a good idea. Likewise, `$announcement->getId()` should be passed through `|escape`, not `|strip_unsafe_html`.

Reply:
URL construction should be done with the `{url ...}` Smarty function; here the host, port, etc. are hard-coded, which will be incorrect.

Reply:
`View` is hard-coded in English; it's best to localize all language.